test_journal.py

```diff
@@ -279,3 +281,133 @@ def test_hacker_cannot_create(app):
     assert response.status_code == 200
     actual = response.body
     assert INPUT_BTN not in actual    # ensure that hackers get redirected
+
+
+# Issue 1 (detail)
+
+def test_view_unit_test_for_permalink():
+    form_str = '<form action="{{ request.route_url(%s) }}" method="get">' % "'detail'"
+    with open('templates/index.jinja2') as f:
+        home = f.read()
+    assert form_str in home
```

Comment:
this is a very odd test.  You appear to be verifying that your template looks like what you expect it to look like.  Generally you don't do that.  Instead you test the result of rendering the template to HTML.  This doesn't really demonstrate more than the fact that you have written your template code to match this test value.